spec/features/variable_drafts/create_variable_draft_with_associated_collection_spec.rb

```diff
@@ -0,0 +1,176 @@
+        expect(page).to have_no_css('.variable-form')
+      end
+
+      it 'notifies the user that the concept supplied was not found' do
+        expect(page).to have_content("No matches were found for C1-INVALID")
+      end
+    end
+
+    context 'when the current provider context is not congruent with the supplied concept-id' do
+      before do
+        fill_in 'associated_collection_id', with: not_current_provider_concept_id
+        click_on 'Create New Record'
+      end
+
+      it 'does not navigate to the new_variable_draft_path' do
+        expect(page).to have_no_content('Variable Information')
+        expect(page).to have_no_css('.umm-form')
+        expect(page).to have_no_css('.variable-form')
+      end
+
+      it 'notifies the user that the provider context needs to be changed' do
+        expect(page).to have_content("Variables can only be associated to collections within the same provider. To create a variable for #{not_current_provider_concept_id} you must change your provider context.") 
+      end
+    end
+
+    context 'when the cmr search for supplied concept-id is unsuccessful' do
+      before do
+        error_body = '{"errors": ["Error searching CMR", "2nd message"]}'
+        error_response = Cmr::Response.new(Faraday::Response.new(status: 401, body: JSON.parse(error_body), response_headers: {}))
+        allow_any_instance_of(Cmr::CmrClient).to receive(:get_collections_by_post).and_return(error_response)
+        fill_in 'associated_collection_id', with: collection_concept_id
+        click_on 'Create New Record'
+      end
+
+      it 'surfaces the first CMR error message' do
+        expect(page).to have_content('Error searching CMR')
+      end
+
+      it 'does not navigate to the new_variable_draft_path' do
+        expect(page).to have_no_content('Variable Information')
+        expect(page).to have_no_css('.umm-form')
+        expect(page).to have_no_css('.variable-form')
+      end
+    end
+
+    context 'when no concept-id is supplied' do
+      before do
+        click_on 'Create New Record'
+      end
+
+      it 'creates a draft with no collection associations' do
+        within '.nav-top' do
+          click_on 'Done'
+        end
+        within '#invalid-draft-modal.eui-modal-content' do
+          click_on 'Yes'
+        end
+        within '.collection-association' do
+          click_on 'Collection Association'
+        end
+        within '#variable-draft-collection-association-table tbody' do
+          expect(page).to have_no_content('CIESIN_SEDAC_ANTHROMES_v2_1700')
+          expect(page).to have_content('No Collection Association found. A Collection must be selected in order to publish this Variable Draft. Each Variable can only be associated with a single Collection.')
+          expect(page).to have_css('tr', count: 1)
+        end
+      end
+    end
+  end
+
+  context "when creating a variable draft and associating a collection from the collection's show page" do
```

Comment:
Same question as manage variables; should these go with collections?

Reply:
Good question. My rationale was that since we are not _affecting_ a collection but rather a variable draft, it should be in the variable drafts folder. But I see where you are coming from, additionally, it is tricky because there isn't necessarily precedent to reference for a situation like this.